### PR TITLE
Generalize Integer type (so it can be changed to e.g., i128)

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
## Description of changes
Generalize Integer type (so it can be changed to e.g., i128)
(To change to i128, some tests for over/underflow behavior also need to be changed)

## Issue #, if available
https://github.com/cedar-policy/cedar/issues/494

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy-core/src/ast/integer.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/// The integer type we use internally.
```
Comment (andrewmwells-amazon) [inline cedar-policy-core/src/ast/integer.rs:1]:
```suggestion
/// The integer types we use (both internally and for parsing).
```

**Review by aaronjeline — APPROVED** [review]
Seems very reasonable. Only wish we could change the text `Long` to `Integer`

**Review by shaobo-he-aws — APPROVED** [review]
LGTM.

File: cedar-policy-core/src/parser/cst_to_ast.rs [hunk 2]
```diff
@@ -3842,6 +3842,7 @@ mod tests {
                 "-9223372036854775809",
                 ToASTErrorKind::IntegerLiteralTooLarge(9223372036854775809),
             ),
+            // This test doesn't fail with an internal representation of i128:
             // Contrary to Rust, this expression is not valid because the
             // parser treats it as a negation operation whereas the operand
             // (9223372036854775808) is too large.
```
Comment (cdisselkoen) [inline cedar-policy-core/src/parser/cst_to_ast.rs:3848]: can we reverse the order here -- first explain why the test has the behavior it does by default (which is the "Contrary to Rust" part) -- then we can have an additional comment about the behavior with i128 (and actually, why doesn't it fail with i128?)

**Review by cdisselkoen — COMMENTED** [review]
I expected to see a change in the lalrpop definition file to use the `InputInteger` type, was that not necessary?

**Pull request merged**